pattern/constraint/thing/HasConstraint.java

```diff
@@ -92,8 +93,8 @@ public String toString() {
     }
 
     @Override
-    public AlphaEquivalence alphaEquals(HasConstraint that) {
-        return AlphaEquivalence.valid().validIfAlphaEqual(attribute, that.attribute);
+    public FunctionalIterator<AlphaEquivalence> alphaEquals(HasConstraint that) {
+        return attribute.alphaEquals(that.attribute);
```

Comment:
Given that we want to move to a constraint-to-constraint comparison for alpha-equivalence, we should include all variables including `owner` in the alpha equals check in each constraint

So we should aim that:
Variable knows how to do alpha-equals to another Variable
Constraint<T> can do alpha-equals to only other Constraint<T>

the combination of alpha-equals in general we do at a higher level - for now in Concludable, in the future maybe in Conjunction :)

When reflecting this change we can simplify the concludable `alphaEquals` methods to not check `owner` explicitly